src/aurora.activity-stream.ts

```diff
@@ -0,0 +1,257 @@
+   */
+  readonly clusterId: string;
+  /**
+   * The identifier of the kms key to be used to encrypt the stream.
+   * Should be the same key used to encrypt the cluster.
+   */
+  readonly kmsKeyId: string;
+}
+
+enum CfnStatus {
+  SUCCESS = 'SUCCESS',
+  FAILED = 'FAILED',
+}
+
+export interface IsCompleteResult {
+  IsComplete: boolean;
+}
+
+export enum CfnRequestType {
+  CREATE = 'Create',
+  UPDATE = 'Update',
+  DELETE = 'Delete',
+}
+
+export interface IsCompleteEvent extends awsLambda.CloudFormationCustomResourceEventCommon {
+  PhysicalResourceId: string;
+  RequestType: CfnRequestType;
+}
+
+interface CreateData {
+  PhysicalResourceId: string;
+}
+
+interface CreateResultProps {
+  Data?: CreateData;
+  PhysicalResourceId: string;
+  ReasonPrefix?: string;
+  Status: CfnStatus;
+}
+
+interface DeleteResultProps {
+  Data?: CreateData;
+  ReasonPrefix?: string;
+}
+
+export class Methods {
+  /**
+   * Given a clusterId, return the ARN of the Aurora Cluster.
+   * @param clusterId
+   * @returns
+   */
+  async getClusterArn(clusterId: string): Promise<string> {
+    const rds = new awsSdk.RDS();
+    const describe = await rds
+      .describeDBClusters({ Filters: [{ Name: 'db-cluster-id', Values: [clusterId] }] })
+      .promise();
+
+    if (!describe.DBClusters || describe.DBClusters.length < 1) {
+      return 'Not found';
+    }
+    if (describe.DBClusters.length > 1) {
+      return 'Multiple clusters found';
+    }
+    return describe.DBClusters[0].DBClusterArn || 'Undefined';
+  }
+
+  /**
+   * Turn on the ActivityStream for the given cluster.
+   * @param event
+   * @param logStreamName
+   * @param invokedFunctionArn
+   * @returns
+   */
+  public async onCreate(
+    event: awsLambda.CloudFormationCustomResourceCreateEvent,
+    logStreamName: string,
+    invokedFunctionArn: string,
+  ): Promise<awsLambda.CloudFormationCustomResourceResponse> {
+    const resultFactory = (props: CreateResultProps): awsLambda.CloudFormationCustomResourceResponse => {
+      return {
+        ...props,
+        LogicalResourceId: event.LogicalResourceId,
+        Reason: `${props?.ReasonPrefix} see also ${logStreamName}`,
+        RequestId: event.RequestId,
+        StackId: event.StackId,
+      };
+    };
+    console.log(`onCreate event: ${JSON.stringify(event)}`);
+
+    const ResourceArn = await this.getClusterArn(event.ResourceProperties.clusterId);
+    if (!ResourceArn.startsWith('arn:aws:rds:')) {
+      return resultFactory({
+        PhysicalResourceId: 'none',
+        ReasonPrefix: ResourceArn,
+        Status: CfnStatus.FAILED,
+      });
+    }
+    console.log(`dbClusterArn: ${ResourceArn}`);
+
+    const rds = new awsSdk.RDS();
+    const result = await rds
+      .startActivityStream({
+        ResourceArn,
+        KmsKeyId: event.ResourceProperties.kmsKeyId,
+        Mode: 'async',
+        ApplyImmediately: true,
+      })
+      .promise();
```

Comment:
Does a call to `.promise()` need to happen here given the `await` keyword?

Reply:
Yes. We need to `.promise()` in order to get the actual result.